packages/project/src/db/styles.ts

```diff
@@ -34,89 +29,108 @@ const assetsLoader = new DataLoader<string, Asset | undefined>(
 );
 
 /**
- * Use zod + DataLoader to load/format assets from the Assets table.
+ * Use DataLoader to load/format assets from the Assets table.
  */
-const ImageAssetDbOut = z.object({
-  type: z.literal("asset"),
-  value: z
-    .string()
-    .uuid()
-    .transform(async (assetId) => {
-      const asset = await assetsLoader.load(assetId);
-      if (asset === undefined) {
-        warnOnce(true, `Asset with assetId "${assetId}" not found`);
-        return;
-      }
-
-      return formatAsset(asset);
-    }),
-});
-
-const ImageValueDbOut = z.object({
-  type: z.literal("image"),
-  value: z
-    .array(ImageAssetDbOut)
-    // an Asset can be not present in DB, skip it.
-    .transform((assets) => assets.filter((asset) => asset.value !== undefined)),
-});
+const loadAsset = async (assetId: string) => {
+  const asset = await assetsLoader.load(assetId);
+  if (asset === undefined) {
+    warnOnce(true, `Asset with assetId "${assetId}" not found`);
+    return;
+  }
 
-const StyleValueDbOut = z.union([SharedStyleValue, ImageValueDbOut]);
+  return formatAsset(asset);
+};
 
-const StylesItemDbOut = z.object({
-  breakpointId: z.string(),
-  instanceId: z.string(),
-  // @todo can't figure out how to make property to be enum
-  property: z.string() as z.ZodType<StyleProperty>,
-  value: StyleValueDbOut,
-});
+const loadValue = async (styleValue: StoredStyles[number]["value"]) => {
+  if (styleValue.type === "image") {
+    return {
+      type: "image" as const,
+      value: (
+        await Promise.all(
+          styleValue.value.map(async (item) => {
```

Comment:
a little too much nesting, makes it harder to read, would put it as a variable above

Reply:
Got rid of promises